packages/components/src/utils/has-prop-value-changed.ts

```diff
@@ -0,0 +1,29 @@
+import { isObject } from './jss';
+
+export const hasPropValueChanged = (newVal: unknown, oldVal: unknown): boolean => {
+  if (typeof newVal !== 'object' || typeof oldVal !== 'object') {
+    // primitive types
+    return newVal === oldVal;
+  } else if (Array.isArray(newVal) && Array.isArray(oldVal)) {
+    // type array
+    return newVal.length === oldVal.length && newVal.every((item) => oldVal.includes(item));
```

Comment:
No need to wrap in another function 
```suggestion
    return newVal.length === oldVal.length && newVal.every(oldVal.includes);
```